### PR TITLE
n8n-4344-insert-and-update-operations-post-null-values-MongoDB-node

### PR DESCRIPTION
**Pull request by michael-radency**

### CODE REVIEW

File: packages/nodes-base/nodes/MongoDb/MongoDb.node.ts [hunk 1]
```diff
@@ -167,45 +285,27 @@ export class MongoDb implements INodeType {
...
-				handleDateFieldsWithDotNotation(updateItems, options.dateFields as string);
-			}
+			const updateItems = prepareItems(items, fields, updateKey, useDotNotation, dateFields);
 
 			for (const item of updateItems) {
 				try {
-					if (item[updateKey] === undefined) {
-						continue;
+					const filter = { [updateKey]: item[updateKey] };
+					if (filter[updateKey] === '_id') {
```
Comment (krynble) [inline packages/nodes-base/nodes/MongoDb/MongoDb.node.ts:305]: This comparison is wrong, we should change against `updateKey` and not `filter[updateKey]`

File: packages/nodes-base/nodes/MongoDb/MongoDb.node.ts [hunk 2]
```diff
@@ -167,45 +285,27 @@ export class MongoDb implements INodeType {
...
+						filter[updateKey] = new ObjectID(item[updateKey] as string);
 					}
 
-					const filter: { [key: string]: string | ObjectID } = {};
-					filter[updateKey] = item[updateKey] as string;
-					if (updateKey === '_id') {
-						filter[updateKey] = new ObjectID(filter[updateKey]);
-						delete item['_id'];
```
Comment (krynble) [inline packages/nodes-base/nodes/MongoDb/MongoDb.node.ts:207]: Removign the deletion of `_id` is causing the update operation to fail

File: packages/nodes-base/nodes/MongoDb/MongoDb.node.ts [hunk 3]
```diff
@@ -124,25 +170,97 @@ export class MongoDb implements INodeType {
...
+			const updateOptions = (this.getNodeParameter('upsert', 0) as boolean)
+				? { upsert: true }
+				: undefined;
+
+			const updateItems = prepareItems(items, fields, updateKey, useDotNotation, dateFields);
+
+			for (const item of updateItems) {
+				try {
+					const filter = { [updateKey]: item[updateKey] };
+					if (filter[updateKey] === '_id') {
```
Comment (krynble) [inline packages/nodes-base/nodes/MongoDb/MongoDb.node.ts:195]: The comparison here is also wrong, making the update based on `_id` to not work

File: packages/nodes-base/nodes/MongoDb/MongoDb.node.ts [hunk 4]
```diff
@@ -124,25 +170,97 @@ export class MongoDb implements INodeType {
...
```
Comment (krynble) [inline packages/nodes-base/nodes/MongoDb/MongoDb.node.ts:234]: Similar issue here

**Review by krynble — CHANGES_REQUESTED** [review]

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.193.2`

**Pull request merged**